tools/image-info

```diff
@@ -2562,9 +2562,9 @@ def append_partitions(report, device, loctl):
     with contextlib.ExitStack() as cm:
         # open each partition as a loop device
         filesystems = {}
-        for part in partitions:
+        for pidx, part in enumerate(sorted(partitions,  key=lambda p: p["start"]), start=1):
             start, size = part["start"], part["size"]
-            dev = cm.enter_context(loop_open(loctl, device, offset=start, size=size))
+            dev = f"{device}p{pidx}"
             read_partition(dev, part)
```

Comment:
This all looks OK to me. However I suggest to move the two lines:
```python
            dev = f"{device}p{pidx}"
            read_partition(dev, part)
```
into the `if partition_is_lvm(part):` block, otherwise the `read_partition()` will be called twice for non-LVM partitions.

Reply:
The way it works now, it's kinda needed.  We need to do `read_partition()` on `part` before checking if it's LVM.  And then if it's not, we change the `dev` and have to re-read it.  It feels redundant, but it's necessary, I think.

Reply:
Looking at CI now, seems this change might have broken other image types.  Worked on my local tests but I didn't cover all cases.

Reply:
> The way it works now, it's kinda needed. We need to do `read_partition()` on `part` before checking if it's LVM. And then if it's not, we change the `dev` and have to re-read it. It feels redundant, but it's necessary, I think.

ah, I didn't realize that. Makes sense...